test/integration/application_controller_test.rb

```diff
@@ -57,18 +57,19 @@ def test_check_browser
         }
       }
     end
-    assert_redirected_to '/p/login'
+    assert_response :forbidden
     # Check that user session was revoked (because of token authenticity)
     assert_not_nil user.user_sessions.reload[0][:revoked_at]
   end
 
-  test "allowed forgery protection won't destroy session when using API controller" do
+  test "forgery protection is skipped for API requests with access token" do
```

Comment:
should we also test when neither token nor basic auth is used?

Reply:
Done